rust/src/connection/network/channel.rs

```diff
@@ -37,73 +37,42 @@ use crate::{
 
 type ResponseFuture = InterceptorResponseFuture<ChannelResponseFuture>;
 
-pub(super) type PlainTextChannel = InterceptedService<Channel, PlainTextFacade>;
 pub(super) type CallCredChannel = InterceptedService<Channel, CredentialInjector>;
 
 pub(super) trait GRPCChannel:
     GrpcService<BoxBody, Error = TonicError, ResponseBody = BoxBody, Future = ResponseFuture> + Clone + Send + 'static
 {
 }
 
-impl GRPCChannel for PlainTextChannel {}
-
 impl GRPCChannel for CallCredChannel {}
 
-pub(super) fn open_plaintext_channel(address: Address) -> PlainTextChannel {
-    PlainTextChannel::new(Channel::builder(address.into_uri()).connect_lazy(), PlainTextFacade)
-}
-
-#[derive(Clone, Debug)]
-pub(super) struct PlainTextFacade;
-
-impl Interceptor for PlainTextFacade {
-    fn call(&mut self, request: Request<()>) -> StdResult<Request<()>, Status> {
-        Ok(request)
-    }
-}
-
 pub(super) fn open_callcred_channel(
     address: Address,
     credential: Credential,
 ) -> Result<(CallCredChannel, Arc<CallCredentials>)> {
     let mut builder = Channel::builder(address.into_uri());
-    if credential.is_tls_enabled() {
-        builder = builder.tls_config(credential.tls_config().clone().unwrap())?;
-    }
     let channel = builder.connect_lazy();
     let call_credentials = Arc::new(CallCredentials::new(credential));
     Ok((CallCredChannel::new(channel, CredentialInjector::new(call_credentials.clone())), call_credentials))
 }
 
 #[derive(Debug)]
 pub(super) struct CallCredentials {
-    credential: Credential,
-    token: RwLock<Option<String>>,
+    credential: Credential
 }
 
 impl CallCredentials {
     pub(super) fn new(credential: Credential) -> Self {
-        Self { credential, token: RwLock::new(None) }
+        Self { credential }
     }
 
     pub(super) fn username(&self) -> &str {
         self.credential.username()
     }
 
-    pub(super) fn set_token(&self, token: String) {
-        *self.token.write().unwrap() = Some(token);
-    }
-
-    pub(super) fn reset_token(&self) {
-        *self.token.write().unwrap() = None;
-    }
-
     pub(super) fn inject(&self, mut request: Request<()>) -> Request<()> {
         request.metadata_mut().insert("username", self.credential.username().try_into().unwrap());
-        match &*self.token.read().unwrap() {
```

Comment:
All this kind of stuff will be reimplemented in encryption, right?

Reply:
No, the removal of this bit is the authentication protocol simplification that I mentioned last week. They're no longer needed!